src/hdmf/container.py

```diff
@@ -228,6 +228,15 @@ def get_ancestor(self, **kwargs):
 
     @property
     def fields(self):
+        '''
+        Subclasses use this class attribute to add properties to autogenerate.
+        `fields` allows for lists and for dicts with the keys {'name', 'child', 'required_name', 'doc', 'settable'}.
+        1. name: The name of the field property
+        2. child: A boolean value to set the parent/child relationship between the field property and the container.
+        3. required_name: The name the field property must have such that `name` matches `required_name`.
+        4. doc: Documentation of the field property
+        5. settable: TBD
+        '''
```

Comment:
``settable`` refers to whether a setter function will be added for the field. If ``true`` a setter function is created so that the field can be changed after creation. If false, then the field is private and cannot be set after init.